### DISCUSSION TITLE
https://notes.alexkehayias.com/typescript-records-with-enum-keys-are-exhaustively-checked/

### DISCUSSION THREAD

**Discussion by giscus** [alexkehayias/note-comments#7] [Announcements]
# https://notes.alexkehayias.com/typescript-records-with-enum-keys-are-exhaustively-checked/

In TypeScript, a Record type used with an enum for keys is exhaustively checked.

https://notes.alexkehayias.com/typescript-records-with-enum-keys-are-exhaustively-checked/

**Comment by oriati** [comment]
and is there a nice way to create a type (e.g. OpenStatusIcon) that will use only 2 statuses from the Status enum?

**Reply by alexkehayias** [reply]
Should be able to that with a union type. 

For example:

```
enum ExampleEnum {
  A=1,
  B=2,
  C=3,
}

// A record that requires two of the three enum variants
const i: Record<ExampleEnum.A | ExampleEnum.B, string> = {
  [ExampleEnum.A]: "foo",
  [ExampleEnum.B]: "bar",
}

// This will throw a compile error
const i: Record<ExampleEnum.A | ExampleEnum.B, string> = {
  [ExampleEnum.A]: "foo",
}
```
